anon/elgam.go

```diff
@@ -0,0 +1,82 @@
+package anon
+
+import (
+	"bytes"
+	"errors"
+	"crypto/cipher"
+	"github.com/dedis/crypto/abstract"
+)
+
+type basicSig struct {
+	C0 abstract.Secret
+	S  abstract.Secret
+}
+
+func signH1preElGam(suite abstract.Suite, message []byte) []byte {
+	H := suite.Hash()
+	H.Write(message)			
+	return H.Sum(nil)
+}
+
+func signH1ElGam(suite abstract.Suite, H1pb []byte, PG abstract.Point) abstract.Secret {
```

Comment:
I don't see why this two-stage hash is needed; why not just include the message rather than the hash of the message in this second one?